daemon/api_debug_raa.go

```diff
@@ -0,0 +1,106 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2025 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package daemon
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/snapcore/snapd/overlord/state"
+	"github.com/snapcore/snapd/sandbox/cgroup"
+	"github.com/snapcore/snapd/snap"
+)
+
+type raaInfo struct {
+	MonitoredSnaps    map[string]monitoredSnapInfo    `json:"monitored-snaps"`
+	RefreshCandidates map[string]refreshCandidateInfo `json:"refresh-candidates"`
+}
+
+type monitoredSnapInfo struct {
+	Pids map[string][]int `json:"pids"`
+}
+
+type refreshCandidateInfo struct {
+	Revision  snap.Revision `json:"revision"`
```

Comment:
This could use an `"omitzero"` although I think it's a recent-ish addition so we can rely on it